pkg/util/ansible_execution.go

```diff
@@ -42,11 +43,16 @@ func AnsibleExecution(
 	obj client.Object,
 	service *dataplanev1.OpenStackDataPlaneService,
 	sshKeySecret string,
```

Comment:
sshKeySecret needs to be updated to handled the same way as inventorySecrets. As each NodeSet could have a different sshKeySecret, we need to pass those in as a list here, and then mount each sshKeySecret into the pod.

you'll likely have to set an ansible group var somehow to make sure each nodeset uses the right SSH key.